### PR TITLE
Reference Data

### PR DESCRIPTION
**Pull request by diogo-s4ntos**
Create branch and Reference Data page.

### CODE REVIEW

**Review by jonathan-osorio — APPROVED** [review]

File: src/app/reference-data/reference-data.component.html [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                        </select>
+                        </section>
+                        <section class="modal-section">
+                            <p><strong>Description:</strong></p>
+                            <input type="text" class="form-control" placeholder="Description..." aria-label="Username" aria-describedby="basic-addon1">
+                        </section>
+                    </div>
+                    <div></div>
+                    <div class="modal-footer">
+                        <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">No</button>
```
Comment (Silva096) [inline src/app/reference-data/reference-data.component.html:44]: Modal buttons should be something like: "Save changes" and "Cancel"

**Review by pmzpro — APPROVED** [review]

**Review by jonathan-osorio — APPROVED** [review]

**Review by Silva096 — CHANGES_REQUESTED** [review]
Bootstrap implementation seems incorrect.
Can you give a try to NgbModule?

File: src/app/app.module.ts [hunk 2]
```diff
@@ -6,11 +6,20 @@ import { AppComponent } from './app.component'
 import { HomeComponent } from './home/home.component'
 import { RouterLink, RouterLinkActive, RouterOutlet, provideRouter } from '@angular/router'
 import routerConfig from './app.routes'
+import { NgbModule } from '@ng-bootstrap/ng-bootstrap'
+import { PropertyManagementComponent } from './property-management/property-management.component'
```
Comment (Silva096) [inline src/app/app.module.ts:10]: why is this component being added here (PropertyManagementComponent) ? Is it needed?

File: src/app/modals/reference-data-modal/reference-data-modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+@Component({
+  selector: 'app-reference-data-modal',
+  templateUrl: './reference-data-modal.component.html',
+  styleUrl: './reference-data-modal.component.css'
+})
+export class ReferenceDataModalComponent {
+  form = referenceDataForm;
+
+  constructor(public activeModal: NgbActiveModal){
+    this.form;
```
Comment (Silva096) [inline src/app/modals/reference-data-modal/reference-data-modal.component.ts:13]: this is not needed.

File: src/app/app.module.ts (hunk 2 above)
Comment (diogo-s4ntos) [reply src/app/app.module.ts]: Done.

File: src/app/modals/reference-data-modal/reference-data-modal.component.ts (hunk 3 above)
Comment (diogo-s4ntos) [reply src/app/modals/reference-data-modal/reference-data-modal.component.ts]: Done.

File: src/styles.css [hunk 4]
```diff
@@ -12,4 +12,8 @@ body {
   --secondary-color: #8B89E6;
   --accent-color: #e8e7fa;
   --shadow-color: #E8E8E8;
+}
+
+.modal-content{
+  border: none !important;
```
[the remark this answers is not in the document]
Comment (diogo-s4ntos) [reply src/styles.css:18]: Done.

File: src/app/modals/reference-data-modal/reference-data-modal.component.html [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<div class="modal d-block">
+    <div class="modal-dialog">
+        <div class="modal-content position-fixed top-0 end-0" id="modal-content">
```
[the remark this answers is not in the document]
Comment (diogo-s4ntos) [reply src/app/modals/reference-data-modal/reference-data-modal.component.html:3]: "d-block" is used to show the modal.

[the remark this answers is not in the document]
Comment (diogo-s4ntos) [reply src/app/modals/reference-data-modal/reference-data-modal.component.html:3]: "d-block" is used to show the modal.

File: src/app/reference-data/reference-data.component.html [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+            <h5>Type:</h5>
+            <select class="form-select" aria-label="Default select example">
+                <option selected>Select Reference Data</option>
+                <option value="1">Cities</option>
+                <option value="2">Typology</option>
+                <option value="3">Realestate Type</option>
+                <option value="3">Amenities</option>
+            </select>
+        </div>
+        <button type="button" class="btn btn-primary" data-toggle="modal" (click)="openModal()">
```
Comment (Silva096) [inline src/app/reference-data/reference-data.component.html:14]: data-toggle="modal" is not needed

**Review by Silva096 — APPROVED** [review]

**Pull request merged**